pkg/controllers/readiness/wellknown_ready_controller.go

```diff
@@ -0,0 +1,351 @@
+					Type:    "WellKnownNotReadyProgressing",
+					Status:  operatorv1.ConditionTrue,
+					Reason:  "RouteCheckFailed",
+					Message: fmt.Sprintf("Route %q is failing health check: %v", route.Name, err),
+				})
+				foundConditions = append(foundConditions, operatorv1.OperatorCondition{
+					Type:    "WellKnownNotReadyAvailable",
+					Status:  operatorv1.ConditionFalse,
+					Reason:  "RouteCheckFailed",
+					Message: fmt.Sprintf("Route %q is failing health check: %v", route.Name, err),
+				})
+			}
+			if len(endpointsMessage) > 0 {
+				foundConditions = append(foundConditions, operatorv1.OperatorCondition{
+					Type:    "WellKnownNotReadyProgressing",
+					Status:  operatorv1.ConditionTrue,
+					Reason:  "EndpointCheckFailed",
+					Message: fmt.Sprintf("Route %q endpoints failing health check: %v", route.Name, err),
+				})
+				foundConditions = append(foundConditions, operatorv1.OperatorCondition{
+					Type:    "WellKnownNotReadyAvailable",
+					Status:  operatorv1.ConditionFalse,
+					Reason:  "EndpointCheckFailed",
+					Message: fmt.Sprintf("Route %q endpoints failing health check: %v", route.Name, err),
+				})
+			}
+		}
+	}
+
+	updateConditionFuncs := []v1helpers.UpdateStatusFunc{}
+	for _, conditionType := range knownConditionNames.List() {
+		// clean up existing foundConditions
+		updatedCondition := operatorv1.OperatorCondition{
+			Type:   conditionType,
+			Status: operatorv1.ConditionFalse,
+		}
+		if strings.HasSuffix(conditionType, "Available") {
+			updatedCondition.Status = operatorv1.ConditionTrue
+		}
+		if condition := v1helpers.FindOperatorCondition(foundConditions, conditionType); condition != nil {
+			updatedCondition = *condition
+		}
+		updateConditionFuncs = append(updateConditionFuncs, v1helpers.UpdateConditionFn(updatedCondition))
+	}
+	if _, _, err := v1helpers.UpdateStatus(c.operatorClient, updateConditionFuncs...); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *wellKnownReadyController) getRoute() (*routev1.Route, []operatorv1.OperatorCondition) {
+	// route is a pre-requirement for this sync
+	// if route does not exists, do nothing and wait
+	route, err := c.routeLister.Routes("openshift-authentication").Get("oauth-openshift")
+	if err != nil && os.IsNotExist(err) {
+		return nil, nil
+	}
+	if err != nil {
+		return nil, []operatorv1.OperatorCondition{
+			{
+				Type:    "WellKnownRouteDegraded",
+				Status:  operatorv1.ConditionTrue,
+				Reason:  "RouteGetFailed",
+				Message: fmt.Sprintf("Unable to get oauth-openshift route: %v", err),
+			},
+		}
+	}
+	return route, nil
+}
+
+func (c *wellKnownReadyController) getAuthConfig() (*configv1.Authentication, []operatorv1.OperatorCondition) {
+	operatorConfig, err := c.authLister.Get("cluster")
+	if err != nil && os.IsNotExist(err) {
+		return nil, nil
+	}
+	if err != nil {
+		return nil, []operatorv1.OperatorCondition{
+			{
+				Type:    "ReadinessControllerDegraded",
```

Comment:
wrong name (wouldn't be unset by your update all conditions logic)

Reply:
hups, good catch!